controller/controller_test.go

```diff
@@ -36,6 +40,8 @@ func TestRun(t *testing.T) {
 	go ctrl.Run(ctx)
 
 	retry.Run(t, func(r *retry.R) {
+		mutex.Lock()
+		defer mutex.Unlock()
 		require.True(r, lister.nsReconciled)
```

Comment:
The issue here is the `go ctrl.Run()` is concurrently writing the fields that are read here. I opted for just a single global mutex for simplicity here.